### DISCUSSION TITLE
Azure IoT DPS

### DISCUSSION THREAD

**Discussion by Shotariusii** [Azure-Samples/iot-middleware-freertos-samples#331] [Q&A] [answered]
Hello, I've successfully managed to connect to the IoT hub, but I have problems with DPS service. I want to register device on the hub through the DPS service. I've created DPS service, added group enrollment and existing hub, but when device establishes MQTT connection to DPS server and tries to subscribe to the MQTT topic Azure SDK returns this massage:  "client registration failed with error 401000, "Unauthorized" ".  I've checked parameters : endpoint, id scope, symmetric key, they are all correct. What can be the other reason for this error ?

**Comment by danewalton-msft** [comment, marked as answer]
Is your symmetric key the group enrollment key or a derived shared access key?

https://learn.microsoft.com/en-us/azure/iot-dps/how-to-legacy-device-symm-key?tabs=linux%22%20%5Cl%20%22derive-a-device-key&pivots=programming-language-ansi-c#derive-a-device-key

**Comment by Shotariusii** [comment]
thats right, the problem was i used enrollment key, not derived key,I have
already changed it.thanks anyway
